beacon-chain/db/pending_deposits.go

```diff
@@ -0,0 +1,85 @@
+package db
+
+import (
+	"context"
+	"math/big"
+
+	"github.com/opentracing/opentracing-go"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+
+	pb "github.com/prysmaticlabs/prysm/proto/beacon/p2p/v1"
+)
+
+var (
+	depositsCount = promauto.NewGauge(prometheus.GaugeOpts{
+		Name: "beacondb_pending_deposits",
+		Help: "The number of pending deposits in the beaconDB in-memory database",
+	})
+)
+
+// Container object for holding the deposit and a reference to the block in
+// which the deposit transaction was included in the proof of work chain.
+type depositContainer struct {
+	deposit *pb.Deposit
+	block   *big.Int
+}
+
+// InsertPendingDeposit into the database. If deposit or block number are nil
+// then this method does nothing.
+func (db *BeaconDB) InsertPendingDeposit(ctx context.Context, d *pb.Deposit, blockNum *big.Int) {
```

Comment:
Should we debug log something if the deposit or block number is nil?

Reply:
Done